components/merkledb/src/macros.rs

```diff
@@ -57,3 +57,22 @@ macro_rules! impl_object_hash_for_binary_value {
         )*
     };
 }
+
+#[macro_export]
+macro_rules! impl_binary_value_for_message {
+    ($( $type:ty ),*) => {
+        $(
+            impl BinaryValue for $type {
+                fn to_bytes(&self) -> Vec<u8> {
```

Comment:
Does exist any reason why we don't return `Result<Vec<u8>, failure::Error>` here ?